src/libraries/Native/Unix/System.Globalization.Native/pal_collation.c

```diff
@@ -568,15 +568,17 @@ static const int32_t GetSearchIteratorUsingCollator(
     usearch_setText(*pSearchIterator, lpSource, cwSourceLength, &err);
     if (!U_SUCCESS(err))
     {
-        int32_t r = RestoreSearchHandle(pSortHandle, *pSearchIterator, options);
+        int32_t r;
+        r = RestoreSearchHandle(pSortHandle, *pSearchIterator, options);
```

Comment:
Just curious, was it causing an error due to assignment and declaration on the same line?

Reply:
It produce unused variable warning/error in release builds.

Reply:
Another way to "fix this" would be to add `(void)r;`. The libraries shims use both patterns.

Reply:
Ah makes sense. Another `(void)` pattern we used in runtime repo is dropping the variable altogether, e.g.:

```c++
(void)RestoreSearchHandle(pSortHandle, *pSearchIterator, options);
```

Reply:
This works only if it is truly unused. It is used here for debug-only assert.